src/mappings/remark.ts

```diff
@@ -3,49 +3,113 @@ import { CallHandlerContext } from '../processor'
 import events from '../types/events'
 import { config } from '../config'
 import { TokenTransfer, Transfer } from '../model'
+import { Call } from '@subsquid/substrate-processor'
 
-export async function handleSystemRemark(ctx: CallHandlerContext) {
+export async function handleBatchWithRemark(ctx: CallHandlerContext) {
     let origin = codec(config.prefix).encode(ctx.call.origin.value.value)
 
-    let matchingBalanceTransfer = await ctx.store.findOne(Transfer, {
-        where: {
-            from: origin,
-            blockNumber: ctx.call.block.height,
-        },
-    })
-
-    let matchingTokenTransfer = await ctx.store.findOne(TokenTransfer, {
-        where: {
-            from: origin,
-            blockNumber: ctx.call.block.height,
-        },
-    })
-
-    // Error if found more than one on the union of both token and balances, or none
-    if (matchingBalanceTransfer && matchingTokenTransfer) {
-        ctx.log.info(
-            `Error when finding the correct transfer for remark with id ${ctx.call.id}. Inconsistent number of transfers for user: ${origin}`
-        )
+    // filter out batch calls where there one transfer and
+    // one remark only
+    if (!hasSystemWithRemark(ctx.call)) {
         return
     }
+    const calls = ctx.call.args.calls
+    const kinds = calls.map((call: any) => call.__kind)
+    if (kinds.length != 2) return
+
+    const remarkCall = calls.find((call: any) => call.__kind === 'System')
```

Comment:
Maybe we should explicitly check if it is a `remark` here, to rule out problems in edge cases where we have multiple different `System` extrinsics inside the batch.

Reply:
The call to `hasSystemWtihRemark` above should be checking that. But I will modify it so it checks if it is only one that is true.